cardano_node_tests/tests/tests_conway/test_drep.py

```diff
@@ -527,6 +575,315 @@ def test_no_witness_register_and_retire(  # noqa: C901
+        helpers.write_json(out_file=drep_metadata_file, content=drep_metadata_content)
+        drep_metadata_hash = cluster.g_conway_governance.drep.get_metadata_hash(
+            drep_metadata_file=drep_metadata_file
+        )
+        deposit_amt = cluster.conway_genesis["dRepDeposit"]
+        drep_keys = cluster.g_conway_governance.drep.gen_key_pair(
+            key_name=temp_template, destination_dir="."
+        )
+        drep_id = cluster.g_conway_governance.drep.get_id(
+            drep_vkey_file=drep_keys.vkey_file,
+            out_format="hex",
+        )
+        reqc.cli090.start(url=helpers.get_vcs_link())
+        # Obtain drep registration certificate
+        reg_cert = cluster.g_conway_governance.drep.gen_registration_cert(
+            cert_name=temp_template,
+            deposit_amt=deposit_amt,
+            drep_vkey_file=drep_keys.vkey_file,
+            drep_metadata_url=drep_metadata_url,
+            drep_metadata_hash=drep_metadata_hash,
+            destination_dir=".",
+        )
+        tx_files_reg = clusterlib.TxFiles(
+            certificate_files=[reg_cert],
+            signing_key_files=[payment_addr.skey_file, drep_keys.skey_file],
+        )
+
+        # Submit drep registration certificate
+        clusterlib_utils.build_and_submit_tx(
+            cluster_obj=cluster,
+            name_template=f"{temp_template}_reg",
+            src_address=payment_addr.address,
+            submit_method=submit_method,
+            use_build_cmd=use_build_cmd,
+            tx_files=tx_files_reg,
+            deposit=deposit_amt,
+        )
+
+        # Wait for some blocks and again submit drep registration certificate
+        cluster.wait_for_new_block(new_blocks=2)
+
+        # Expecting error for both cases as drep is already registered
+        if submit_method == "api":
+            with pytest.raises(submit_api.SubmitApiError) as excinfo_submit:
+                clusterlib_utils.build_and_submit_tx(
+                    cluster_obj=cluster,
+                    name_template=f"{temp_template}_reg2",
+                    src_address=payment_addr.address,
+                    submit_method=submit_method,
+                    use_build_cmd=use_build_cmd,
+                    tx_files=tx_files_reg,
+                    deposit=deposit_amt,
+                )
+            err_msg = str(excinfo_submit.value)
+            err_msg = err_msg.replace("\\", "")
+        else:
+            with pytest.raises(clusterlib.CLIError) as excinfo_cli:
+                clusterlib_utils.build_and_submit_tx(
+                    cluster_obj=cluster,
+                    name_template=f"{temp_template}_reg2",
+                    src_address=payment_addr.address,
+                    submit_method=submit_method,
+                    use_build_cmd=use_build_cmd,
+                    tx_files=tx_files_reg,
+                    deposit=-deposit_amt,
+                )
+            err_msg = str(excinfo_cli.value)
```

Comment:
This can be simplified to handle both `SubmitApiError` and `CLIError` in single `pytest.raises` context.

Reply:
Initially, submit api and cli were giving two different error strings and mypy was complaining about types. 

Now It has been updated to handle both errors in single pytest.raises with common error message between both.